pkg/doc/signature/jsonld/processor.go

```diff
@@ -26,23 +26,31 @@ var logger = log.New("aries-framework/json-ld-processor")
 // normalizeOpts holds options for canonicalization of JSON LD docs
 type normalizeOpts struct {
 	removeInvalidRDF bool
-	removeRDFMatches []string
+	documentLoader   ld.DocumentLoader
+	extraContexts    []string
 }
 
-// CanonicalizationOpts are the options for canonicalization of JSON LD docs
-type CanonicalizationOpts func(opts *normalizeOpts)
+// ProcessorOpts are the options for JSON LD operations on docs (like canonicalization or compacting).
+type ProcessorOpts func(opts *normalizeOpts)
 
 // WithRemoveAllInvalidRDF option for removing all invalid RDF dataset from normalize document
-func WithRemoveAllInvalidRDF() CanonicalizationOpts {
+func WithRemoveAllInvalidRDF() ProcessorOpts {
 	return func(opts *normalizeOpts) {
 		opts.removeInvalidRDF = true
 	}
 }
 
-// WithRemoveInvalidRDF option for removing RDFs containing any one of the given matches
-func WithRemoveInvalidRDF(matches ...string) CanonicalizationOpts {
+// WithDocumentLoader option is for passing custom JSON-LD document loader.
+func WithDocumentLoader(loader ld.DocumentLoader) ProcessorOpts {
 	return func(opts *normalizeOpts) {
-		opts.removeRDFMatches = matches
+		opts.documentLoader = loader
+	}
+}
+
+// WithExtraContext option is for using extra context when doing JSON-LD operations.
+func WithExtraContext(context ...string) ProcessorOpts {
```

Comment:
I guess this is sometimes called external context: https://w3c.github.io/json-ld-syntax/#interpreting-json-as-json-ld

Reply:
I think this is rather a hack solution: we artificially append extra contexts before doing any JSON-LD operation like compaction or normalization in order to use enlarged JSON-LD vocabularies and be capable of recognizing extra types like `name`, `description` (see https://github.com/w3c-ccg/vc-examples/issues/22), `CredentialStatusList2017`, ...
This approach should be used if we receive JSON-LD to parse and it contains out-of-vocabulary terms we still want to support (e.g. because they were used to build a signature verification data).

Reply:
Agree! It would be more correct.